strapi/src/api/course-statistics/controllers/course-statistics-feedback.ts

```diff
@@ -0,0 +1,37 @@
+import { errorCodes } from "../../../helpers/errorCodes";
+
+interface Feedback {
+  rating: number;
+}
+
+export default {
+    async getAverageCourseFeedback(ctx) {
+      try {
+            const courseId = ctx.params.courseId;
+            const course = await strapi.documents("api::course.course").findOne({
+              documentId: courseId,
+              populate: ["feedbacks"],
+            });
+    
+            if (!courseId) {
+              ctx.response.status = 500;
+              ctx.response.body = { error: errorCodes["E0006"] };
+            }
+    
```

Comment:
The `courseId` check should probably be before it is used.

```suggestion
            const courseId = ctx.params.courseId;
                
            if (!courseId) {
              ctx.response.status = 500;
              ctx.response.body = { error: errorCodes["E0006"] };
            }
            
            const course = await strapi.documents("api::course.course").findOne({
              documentId: courseId,
              populate: ["feedbacks"],
            });
```